src/database/Database.cpp

```diff
@@ -196,9 +196,12 @@ Database::applySchemaUpgrade(unsigned long vers)
         mApp.getHerderPersistence().createQuorumTrackingTable(mSession);
         break;
     case 11:
-        mSession << "DROP INDEX IF EXISTS bestofferindex;";
-        mSession << "CREATE INDEX bestofferindex ON offers "
-                    "(sellingasset,buyingasset,price,offerid);";
+        if (mApp.modeCommitsLedgerTxnRoot())
```

Comment:
I see `modeCommitsLedgerTxnRoot` more as a commit strategy.

I don't think we need to make the code more complicated than it is already: we can create the tables but just never commit to them.
With this toggle, it makes it very error prone as core could be pointed to an existing database that ends up being partially reset if the toggle is set.

Reply:
The problem is that it does not work with the existing mNeverCommittingLedgerTxn, since it's not really a LedgerTxnRoot, but just a regular LedgerTxn. This class explicitly prohibits calling functions such as `dropAccounts` on it.

For the case you described, I don't know if we should rely on any of the testing modes to properly reset the database (currently, the only reason we hit the reset codepath is because we use in-memory db). If we want to reset the DB prior to running a test mode, we could follow the same pattern we use for normal node operation: run `new-db` (in normal mode), then `run`/`catchup` in special mode.